modules/dynamic-plugin-proxy-service.adoc

```diff
@@ -0,0 +1,42 @@
+// Module included in the following assemblies:
+//
+// * web_console/dynamic-plugin/deploy-plugin-cluster.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="dynamic-plugin-proxy-service_{context}"]
+= Plugin service proxy
+
+The plugin Operator creates a deployment on the platform with an HTTP server to host the plugin and expose it using a Kubernetes Service. The HTTP server serves all assets for the plugin, including JSON, JavaScript, CSS, and images. The `Service` must use HTTPS and a service serving certificate. The console backend will proxy the plugin assets from the `Service` using the service certificate authority (CA) bundle. Operators declare that they have a console plugin available by creating a cluster-scoped `ConsolePlugin` resource that includes information about the backend `service` `name`, `port`, and `basePath` used to access all of the plugin's assets.
+
+If a plugin needs an in-cluster service, you can declare a service proxy in its `ConsolePlugin` resource using the `spec.proxy` array field. For each entry you must specify an endpoint and alias of the proxy under the `endpoint` and `alias` fields. For the Service proxy type, the endpoint `type` field will need to be set to `Service` and the `service` must include a `name`, `namespace`, and `port`. For example, `/api/proxy/plugin/helm/helm-charts/releases++?++limit++=++10` is a proxy request path from the `helm` plugin with a `helm-charts` service that lists ten helm releases.
+
+The console backend exposes the `/api/proxy/plugin/<plugin-name>/<proxy-alias>/<request-path>++?++<optional-query-parameters>` endpoint to proxy the communication between plugin and the service. A proxied request uses a service CA bundle by default. The service must use HTTPS.
+
+.Example
+[source,text,subs="+quotes,+macros"]
+----
++`Authorization: Bearer sha256~kV46hPnEYhCWFnB85r5NrprAxggzgb6GOeLbgcKNsH0`+ <1>
```

Comment:
This isn't a part of the ConsolePlugin resource, but an example request header that is sent from console as part of the plugin HTTP request to the target service.